### PR TITLE
[SFPA-36] Category Scene UI Tests

### PR DESCRIPTION
**Pull request by stoikokolev**

### CODE REVIEW

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift [hunk 1]
```diff
@@ -30,6 +31,11 @@ class CategoryItemTableViewCell: UIResizableTableViewCell {
         }
     }
     
+    private func setupAccessibility() {
+        tagsStackView.isAccessibilityElement = true
+        tagsStackView.accessibilityIdentifier = "tagsStackView"
```
Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift:36]: Extract to the Constants struct or create an enum only for the accessibilityIdentifiers

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift [hunk 2]
```diff
@@ -30,6 +31,11 @@ class CategoryItemTableViewCell: UIResizableTableViewCell {
         }
     }
     
+    private func setupAccessibility() {
+        tagsStackView.isAccessibilityElement = true
```
Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift:35]: Check if the default value of 'isAccessibilityElement' isn't true

Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift:35]: Check if the default value of 'isAccessibilityElement' isn't true

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift (hunk 1 above)
Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift:36]: Extract to the Constants struct or create an enum only for the accessibilityIdentifiers

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/ListViewController.swift [hunk 3]
```diff
@@ -48,6 +45,14 @@ class ListViewController: BaseViewController {
         }
     }
     
+    private func setupTableView() {
+        categoriesTableView.register(cellNames: viewModel.reuseIdentifiers)
+        let backgroundView = UIImageView(image: UIImage.background)
+        backgroundView.isAccessibilityElement = true
+        backgroundView.accessibilityIdentifier = "listTableViewBackground"
```
Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/ListViewController.swift:52]: Same as previous comment

File: SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+//  SFPortfolioAppUITests
+//
+//  Created by Stoyko Kolev on 20.08.21.
+//
+
+import Foundation
+
+struct AccessibilityIdentifiers {
+    
+    static let buttonsCell = "Buttons"
```
Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift:12]: Should start with lowercase

File: SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    static let listTableView = "listTableView"
+    static let listTableViewBackground = "listTableViewBackground"
+    static let logo = "logo"
+    static let categoriesNavigationBar = "Categories"
```
Comment (plamenzhl) [inline SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift:16]: Same as previous comment

**Review by plamenzhl — CHANGES_REQUESTED** [review]

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift (hunk 2 above)
Comment (stoikokolev) [reply SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift]: The default value is _false_ for a standard UIKit controls and is removed here for the stackView.

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift (hunk 1 above)
Comment (stoikokolev) [reply SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/CategoryTableViewCell/CategoryItemTableViewCell.swift]: Fixed

File: SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/ListViewController.swift (hunk 3 above)
Comment (stoikokolev) [reply SFPortfolioApp/SFPortfolioApp/Modules/CategoryModule/BaseListScene/ListViewController.swift]: The default value is _false_ for a standard UIKit controls and here for the UIView as not a standard control we need to set it as _true_.

File: SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift (hunk 4 above)
Comment (stoikokolev) [reply SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift]: Fixed with creating two separate structs Label and Identifier.

File: SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift (hunk 5 above)
Comment (stoikokolev) [reply SFPortfolioApp/SFPortfolioAppUITests/AccessibilityIdentifiers.swift]: Fixed

**Review by plamenzhl — APPROVED** [review]

**Pull request merged**